### PR TITLE
Summarization ability

### PR DESCRIPTION
**Pull request by derekchen14**
Not quite ready to merge yet, but making PR so we can easily review

### CODE REVIEW

File: negotiation/evaluate.py [hunk 1]
```diff
@@ -48,8 +48,9 @@
         cuda.set_device(args.gpuid[0])
 
     # Load the model.
+    ckpt_file = args.checkpoint_files[0] if isinstance(args.checkpoint_files, list) else args.checkpoint_files
```
Comment (hhexiy) [inline negotiation/evaluate.py]: just use the list argument (ngargs='+') so that we don't have to check if it's a list

File: negotiation/neural/model_builder.py [hunk 2]
```diff
@@ -210,7 +210,7 @@ def make_base_model(model_opt, mappings, gpu, checkpoint=None):
     kb_embeddings = make_embeddings(model_opt, kb_dict)
     kb_embedder = make_context_embedder(model_opt, kb_embeddings, 'kb')
     # Make decoder.
-    if model_opt.model == 'seq2seq':
+    if model_opt.model in ['seq2seq', 'sum2sum', 'sum2seq']:
```
Comment (hhexiy) [inline negotiation/neural/model_builder.py]: for sum2sum you need a separate tgt_vocab because the vocab is much more smaller right?

File: negotiation/neural/preprocess.py [hunk 3]
```diff
@@ -216,18 +216,18 @@ def _add_utterance(self, agent, utterance, lf=None):
         else:
             lf = []
 
-        if not new_turn:
-            self.token_turns[-1].extend(utterance)
-            self.lfs[-1].extend(lf)
-            self.entities[-1].extend(entities)
-        else:
+        if new_turn:
```
Comment (hhexiy) [inline negotiation/neural/preprocess.py:233]: why did you remove the if not new_turn part?

File: negotiation/neural/preprocess.py [hunk 4]
```diff
@@ -413,6 +401,44 @@ def process_event(self, e, kb):
...
+            for e in ex.events:
+                utterance = self.process_event(e, dialogue.kb)
+                if utterance:
+                    if self.model == "sum2sum":
+                        utterance = self.summarize(utterance)
+                    elif self.model == "sum2seq":
+                        summary = self.summarize(utterance)
+                        # Do we want to add some EOD marker here?
+                        summary.extend(utterance)
+                    dialogue.add_utterance(e.agent, utterance, lf=e.metadata)
```
Comment (hhexiy) [inline negotiation/neural/preprocess.py]: can you add summary separately, i.e. don't overwrite utterance, because for sum2seq you still need the utterance data.

File: negotiation/neural/preprocess.py (hunk 3 above)
Comment (derekchen14) [reply negotiation/neural/preprocess.py:233]: There is still a "new_turn" check, I just reversed the order to avoid the "not"
Since new_turn is True or False, one of those if-else blocks must run, so it should be safe.  Let me know if that's not true.
